cmd/prometheus/main.go

```diff
@@ -233,6 +233,7 @@ func main() {
 	}
 
 	promql.LookbackDelta = time.Duration(cfg.lookbackDelta)
+	promql.SetGlobalEvaluationInterval(time.Duration(config.DefaultGlobalConfig.EvaluationInterval))
```

Comment:
Can we avoid a global variable here (well, you always can, just a question of how much work)? AFAICS it would mainly need to be passed into all places that create an engine, and from the engine into the evaluator?

Btw. in the `promql` package we should not call this `GlobalEvaluationInterval`, since that is a rule-related concept that has nothing to do with PromQL per se. How about `DefaultEvaluationInterval` instead? That's all that PromQL needs to know.

Reply:
As it is passed down through Engine, it would need passing from all the places where we create Engine - that might need access to the global config. I would say it's better if we set it once, and it is used thereafter. If there is any benefit of passing down through Engine, I can do that.

I will change the name.